packages/template-retail-react-app/app/components/product-view-modal/bundle.jsx

```diff
@@ -32,20 +32,31 @@ import {useDerivedProduct} from '@salesforce/retail-react-app/app/hooks'
 const BundleProductViewModal = ({product: bundle, isOpen, onClose, updateCart, ...props}) => {
     const productViewModalData = useProductViewModal(bundle)
     const {variationParams} = useDerivedProduct(bundle)
-
-    const childProductIds = productViewModalData.product?.bundledProductItems
-        ?.map(({productId}) => productId)
-        .join(',')
-    const {data: childProducts, isLoading} = useProducts(
-        {parameters: {ids: childProductIds, allImages: true}},
-        {enabled: Boolean(childProductIds)}
-    )
-
     const childProductRefs = useRef({})
     const [childProductOrderability, setChildProductOrderability] = useState({})
     const [selectedChildProducts, setSelectedChildProducts] = useState([])
-
+    const [selectedBundleQuantity, setSelectedBundleQuantity] = useState(
+        productViewModalData?.product?.quantity
+    )
     const trueIfMobile = useBreakpointValue({base: true, lg: false})
+    const [childProductIds, setChildProductIds] = useState(
+        productViewModalData.product?.bundledProductItems?.map(({productId}) => productId).join(',')
+    )
+
+    useEffect(() => {
+        const productIds = selectedChildProducts.map(({variant}) => variant.productId).join(',')
+        if (productIds?.length > 0 && productIds !== childProductIds) {
+            setChildProductIds(productIds)
+        }
+    }, [selectedChildProducts])
```

Comment:
Typically a state is used in the rendering, but this one is not. So we can simply use a variable instead.

```suggestion
    let childProductIds = productViewModalData.product?.bundledProductItems
        ?.map(({productId}) => productId)
        .join(',')
    const productIds = selectedChildProducts.map(({variant}) => variant.productId).join(',')
    if (productIds?.length > 0 && productIds !== childProductIds) {
        childProductIds = productIds
    }
```